### PR TITLE
Role Clarifications in /mafia dt

### PR DESCRIPTION
**Pull request by ayiadvance**
Created distinctions between:
Role Cop and Role Cop (Classic)
Red Goo and Red Goo (Cracking Idea)
Neapolitan and Vanilla Cop

Also clarified some roles and standardized how Idea Variant roles are listed.

### CODE REVIEW

**Comment by YashGreninja** [comment]
If we are doing distinctions within /mafia dt, can we also implement Silencer and Silencer GI (since those are distinctly different)?

File: server/chat-plugins/mafia-data.js [hunk 1]
```diff
@@ -390,7 +402,7 @@ const roles = {
 	coroner_cracking_idea: `coronercrackingidea`,
 	coronercrackingidea: {
 		name: `Coroner (Cracking Idea)`,
-		id: `coronercrackingidea`,
+		id: `coronerci`,
```
Comment (HoeenCoder) [inline server/chat-plugins/mafia-data.js]: Why is this ID changing? it dosent match the property now.

File: server/chat-plugins/mafia-data.js [hunk 2]
```diff
@@ -530,12 +542,24 @@ const roles = {
...
+	graygooci: `greygoocrackingidea`,
+	gray_goo_ci: `greygoocrackingidea`,
+	gray_goo_cracking_idea: `greygoocrackingidea`,
+	graygoocrackingidea: `greygoocrackingidea`,
+	greygooci: `greygoocrackingidea`,
+	grey_goo_ci: `greygoocrackingidea`,
+	grey_goo_cracking_idea: `greygoocrackingidea`,
+	greygoocrackingidea: {
+		name: `Grey Goo (Cracking Idea)`,
+		id: `greygooci`,
```
Comment (HoeenCoder) [inline server/chat-plugins/mafia-data.js]: ID should not match an alias, but rather the main property eg: `greygoocrackingidea`

File: server/chat-plugins/mafia-data.js [hunk 3]
```diff
@@ -844,9 +868,12 @@ const roles = {
...
 	},
-	red_goo: {
-		name: `Red Goo`,
-		id: `redgoo`,
+	redgooci: `redgoocrackingidea`,
+	red_goo_ci: `redgoocrackingidea`,
+	red_goo_cracking_idea: `redgoocrackingidea`,
+	redgoocrackingidea: {
+		name: `Red Goo (Cracking Idea)`,
+		id: `redgooci`,
```
Comment (HoeenCoder) [inline server/chat-plugins/mafia-data.js]: `id` should match the property

File: server/chat-plugins/mafia-data.js (hunk 1 above)
Comment (HoeenCoder) [inline server/chat-plugins/mafia-data.js]: `id` should be the ID version (all lowercase, no spaces or symbols, keep numbers) of the property (eg: `coronercrackingidea`).

File: server/chat-plugins/mafia-data.js (hunk 2 above)
Comment (HoeenCoder) [inline server/chat-plugins/mafia-data.js]: Same ID issue here.

File: server/chat-plugins/mafia-data.js (hunk 3 above)
Comment (HoeenCoder) [inline server/chat-plugins/mafia-data.js]: Same ID issue here.

File: server/chat-plugins/mafia-data.js [hunk 4]
```diff
@@ -924,6 +951,14 @@ const roles = {
 		id: `silencer`,
 		memo: [`Silencer: During the Night, you may PM the host the name of another player. This player will not be able to talk during the next day. Ask the host whether this player will be able to lynch or not.`],
 	},
+	silencer_gi: `silencergreateridea`,
+	silencergi: `silencergreateridea`,
+	silencer_cracking_idea: `silencergreateridea`,
+	silencergreateridea: {
+		name: `Silencer (Greater Idea)`,
+		id: `silencergi`,
```
Comment (HoeenCoder) [inline server/chat-plugins/mafia-data.js]: Same ID issue here.

File: server/chat-plugins/mafia-data.js [hunk 5]
```diff
@@ -359,10 +359,22 @@ const roles = {
...
+		image: `<img width="75" height="75" src="//${Config.routes.client}/fx/mafia-cop.png"/>`,
+	},
+	role_cop_classic: `rolecopclassic`,
+	rolecopclassic: {
+		name: `Role Cop (Classic)`,
+		id: `rolecopclassic`,
+		memo: [`Role Cop: Each night you can PM the host the name of another player. You will be told their exact role and alignment, or NO RESULT if your investigation failed.`],
+		image: `<img width="75" height="75" src="//${Config.routes.client}/fx/mafia-cop.png"/>`,
+	},
+	vanilla_cop: {
```
Comment (HoeenCoder) [inline server/chat-plugins/mafia-data.js]: Isn't this the same as the neapolitan but under a different name?

Comment (ayiadvance) [reply server/chat-plugins/mafia-data.js]: No, Vanilla Cop checks Vanilla / Not Vanilla, while Neapolitan specifically checks VT / not VT. It's subtle but it has very different uses in setups.

**Pull request merged**